units/src/amount/unsigned.rs

```diff
@@ -175,19 +175,16 @@ impl Amount {
     ///
     /// If the amount is too precise, negative, or greater than 21,000,000.
     pub fn from_str_in(s: &str, denom: Denomination) -> Result<Amount, ParseAmountError> {
-        let (negative, sats) =
+        let (is_neg, sats) =
             parse_signed_to_satoshi(s, denom).map_err(|error| error.convert(false))?;
-        if negative {
-            return Err(ParseAmountError(ParseAmountErrorInner::OutOfRange(
-                OutOfRangeError::negative(),
-            )));
-        }
-        if sats > Self::MAX.to_sat() {
-            return Err(ParseAmountError(ParseAmountErrorInner::OutOfRange(
-                OutOfRangeError::too_big(false),
-            )));
-        }
-        Ok(Self::from_sat_unchecked(sats))
+        let sats = match u64::try_from(sats) {
+            Ok(s) if !is_neg => s,
+            _ =>
+                return Err(ParseAmountError(ParseAmountErrorInner::OutOfRange(
+                    OutOfRangeError::negative(),
+                ))),
+        };
```

Comment:
Yeah, if we returned `SignedAmount` this entire thing would simplify to `if is_neg { return ... }`